src/battle_ai_switch_items.c

```diff
@@ -1785,14 +1785,59 @@ static bool32 CanAbilityTrapOpponent(u16 ability, u32 opponent)
         return FALSE;
 }
 
+static bool32 IsFreeSwitch(bool32 isSwitchAfterKO, u32 battlerSwitchingOut, u32 opposingBattler)
+{
+    bool32 movedSecond = GetBattlerTurnOrderNum(battlerSwitchingOut) > GetBattlerTurnOrderNum(opposingBattler) ? TRUE : FALSE;
```

Comment:
I've mentioned it previously but I'm still skeptical because `GetBattlerTurnOrderNum` can have values from the previous turn which can produce incorrect results. Say the opposing mon went second, used Icy Wind and is faster now. The change won't be reflected here since it updates on the next turn when a battler starts moving.   

Is there any reason we can't compare speed directly with `AI_IsLower`?

Reply:
oh, Ig I misunderstood how this works. I decides what to do after the attack, right?

Reply:
Also I'm sorry if you just got inundated by comments I was writing these on mobile and somehow started a review while writing them lol

Reply:
This is all fine then. No complaints here. Was just a misunderstanding.

Reply:
Okay sweet!

... D'you mind clarifying how you misunderstood so I can make sure it's also good in my head, cuz I'm not entirely sure what your understand was and I want to make sure I also agree this works fine lol

Reply:
I didn't realize that the function is executed after a switch happened and now the AI decides which mon to switch in.

Reply:
OHHHH okay, I think we're on the same page then this sounds okay in that context to me :)